api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+              (TBD: enforce UUID formatif set by the client?)
+            type: string
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          event_type:
+            description: |
+              The EventType of this Event. This is enriched by Nakadi on reception of the Event based on the
+              endpoint where the Producer sent the Event to.
+
+              If provided MUST match the endpoint. Failure to do so will cause rejection of the Event.
+            type: string
+            example: 'pennybags:payment-business-event'
+          occurred_at:
+            description: |
+              Timestamp of creation of the Event generated by the producer. Presence is optional but strongly
+              recommended for producers.
+            type: string
+            format: date-time
+          received_at:
+            type: string
+            description: |
+              Timestamp of the reception of the Event by Nakadi. This is enriched upon reception of the
+              Event.
+              If set by the producer Event will be rejected.
+            format: date-time
+          parent_eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the Event that caused the generation of this Event.
+              Set by the procuder.
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          root_eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the root of the tree of Events (from the relation constructed from
+              parent-eid)
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          flow_id:
+            description: |
+              The flow-id of the producer of this Event. As this is usually a HTTP header, this is enriched
+              from the header into the metadata by Nakadi to avoid clients having to explicitly copy this.
+            type: string
+            example: 'JAh6xH4OQhCJ9PutIV_RYw'
+
+        additionalProperties: false
+
+    required:
+      - metadata
+
+  BusinessEvent:
+    type: object
+    description: |
+      A Business Event.
+
+      Usually represents a status transition in a Business process.
+    allOf:
+      - $ref: '#/definitions/EventMetadata'
+    additionalProperties: true
+
+  DataChangeEventQualifier:
+    type: object
+    description: |
+      Indicators of a `DataChangeEvent`'s referred data type and the type of operations done on them.
+
+    properties:
+      data_type:
+        type: string
+        example: 'pennybags:order'
+      data_op:
+        type: string
+        enum: ['C', 'U', 'D', 'S']
+        description: |
+          The type of operation executed on the entity.
+          * C: Creation
+          * U: Update
+          * D: Deletion
+          * S: Snapshot
+    required:
+      - data_type
+      - data_op
+
+  DataChangeEvent:
```

Comment:
@aikej  I would prefer to have some kind of a convention about the construction of the URI. What I am afraid of, is that making such an URI we would introduce quite a significant limitation on the payload format in generic case. 

But would not be a requirement from the RESTful resource schema prospective, to provide an URI of the primary key in resource itself? It would even most probably will be the value of the filed(s) that would be marked as a primary key for that resource?